middleware/AuthorizationHandler.md

```diff
@@ -13,15 +12,22 @@ A piece of client-side middleware intended to allow the use of an authorization
 - Certain workloads error out when an unexpected header is present in the request. The middlewares should check:
   - If the request URL is a Graph serve endpoint or a custom url provided by the developer, then append or update the authorization header.
   - Else the middleware should delete authorization header.
- 
 
 Take a request object and use authorization provider to add Authorization header.  
 
+### Additional requirements for Kiota clients
+
+- MUST NOT attempt to update the authorization header if it is already present (unless we ran into an authentication challenge). This is to ensure customers can use the same http client for arbitrary requests and with a generated client/request adapter without running into double authentication issues.
+- MUST accept a [BaseBearerAuthenticationProvider](https://github.com/microsoft/kiota-dotnet/blob/main/src/abstractions/authentication/BaseBearerTokenAuthenticationProvider.cs) and use the associated AccessTokenProvider to obtain the access token. This is to ensure the client initialization remains simple.
+- MUST be implemented in the graph core or kiota bundle layer, not in the kiota http or authentication layers. This is to ensure we do not mix concerns in the authentication/http packages.
```

Comment:
Why don't we want this in the Kiota http layer? Not clear on the potential mixed concerns.

Reply:
I think you are correct actually, this could live in HTTP since all we need here is defined in the abstractions layer. Thanks for spotting it, I'll update in a moment.